appserver/payara-appserver-modules/microprofile/metrics/src/main/java/fish/payara/microprofile/metrics/jmx/MetricsMetadataHelper.java

```diff
@@ -108,18 +112,33 @@ public List<MBeanMetadata> registerMetadata(MetricRegistry metricRegistry,
                     continue;
                 }
                 Metric type;
-                MBeanExpression mBeanExpression = new MBeanExpression(beanMetadata.getMBean());
-                switch (beanMetadata.getTypeRaw()) {
-                    case COUNTER:
-                        type = new MBeanCounterImpl(mBeanExpression);
-                        break;
-                    case GAUGE:
-                        type = new MBeanGuageImpl(mBeanExpression);
-                        break;
-                    default:
-                        throw new IllegalStateException("Unsupported type : " + beanMetadata);
+                if (beanMetadata.getMBean() != null) {
+                    MBeanExpression mBeanExpression = new MBeanExpression(beanMetadata.getMBean());
+                    switch (beanMetadata.getTypeRaw()) {
+                        case COUNTER:
+                            type = new MBeanCounterImpl(mBeanExpression);
+                            break;
+                        case GAUGE:
+                            type = new MBeanGuageImpl(mBeanExpression);
+                            break;
+                        default:
+                            throw new IllegalStateException("Unsupported type : " + beanMetadata);
+                    }
+                    metricRegistry.register(beanMetadata, type, tags.toArray(new Tag[tags.size()]));
+                } else {
+                    HealthCheckStatsProvider healthCheck = Globals.getDefaultHabitat().getService(HealthCheckStatsProvider.class, beanMetadata.getService());
```

Comment:
I'd recommend `@Inject`ing a service locator instead, `Globals` lookup should be left for places when we are outside of HK2 control, but this one is a `@Service` already. We're overusing this pattern a lot and then we get random explosions on startup ;)

What is the behavior in error cases? What happens when invalid service name is given?